File: Project-Tech/Backend/unlockit/src/main/java/io/unlockit/model/mongodb/Proposal.java
package io.unlockit.model.mongodb;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Document("proposal")
public class Proposal {

    @Id
    private String id;
    @NotNull(message = "invalid tenant id, cannot be null")
    private String tenantId;
    @NotNull(message = "invalid contract id, cannot be null")
    private String contractId;
    @NotNull(message = "invalid price, cannot be null")
    @Min(1)
    private int originalPrice;
    @Min(1)
    private int proposalPrice;
    private boolean active;
    @NotNull()
    private String status;
}

File: Project-Tech/Hyperledger-Fabric/unlockit-smart-contract/src/model/proposal.ts
/*
  SPDX-License-Identifier: Apache-2.0
*/

import {Object, Property} from 'fabric-contract-api';

@Object()
export class Proposal {
    @Property()
    public id: string;

    @Property()
    public tenantId: string;

    @Property()
    public tenantSignature: string;

    @Property()
    public contractId: string;

    @Property()
    public originalPrice: number;

    @Property()
    public proposalPrice: number;

    @Property()
    public active: boolean;

    @Property()
    public status: string;
}
